tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7624,6 +7624,9 @@ def warn_hlsl_narrowing : Warning<
 def warn_hlsl_sema_minprecision_promotion : Warning <
   "%0 is promoted to %1">,
   InGroup<Conversion>, DefaultWarn;
+def warn_hlsl_ambiguous_literal_shift : Warning <
+  "use a type suffix on literal values, like 'L' or 'U', or a cast, to resolve ambiguity">,
```

Comment:
Message seems to be missing the first part identifying what the warning is about.

How about:
`ambiguous type for bit shift; use a type suffix on literal values, like 'L' or 'U', or a cast`